homework/week1/fetchAndInsertData.js

```diff
@@ -0,0 +1,80 @@
+const fetch = require('node-fetch');
+const mysql = require('mysql');
+const { config, configWithDb } = require('./connection.js');
+const URL_COUNTRIES = 'https://restcountries.eu/rest/v2/all';
+const URL_CITIES =
+  'https://public.opendatasoft.com/api/records/1.0/search/?dataset=worldcitiespop&rows=3000&sort=population';
+
+const connection = mysql.createConnection(configWithDb);
+
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    return await response.json();
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function convertCountryDataToArray(dataJson) {
+  const countries = [];
+
+  dataJson.forEach(country => {
+    countries.push([
+      country.alpha2Code,
+      country.name,
+      country.region,
+      country.capital,
+      country.population,
+      country.area,
+    ]);
+  });
+
+  return countries;
+}
+
+function convertCityDataToArray(dataJson) {
+  const cities = [];
+  dataJson.records.forEach(data => {
+    const city = data.fields;
+    cities.push([city.city, city.country, city.population]);
+  });
+
+  return cities;
+}
+
+async function insertData(data, tableName, columns = []) {
+  if (columns.length > 0) {
+    tableName += `(${columns.join(',')})`;
+  }
+
+  const insertSql = `INSERT INTO ${tableName} VALUES ?`;
+  return new Promise((resolve, reject) => {
+    connection.query(insertSql, [data], (error, result) => {
```

Comment:
Correct me if I'm wrong - don't you need to loop `data` and insert into the table row by row?

Reply:
First of all thank you so much for your review. Insertion can be  made row by row as you said, indeed can also be made as bulk insert with this format: INSERT INTO table('col1','col2','col3') VALUES ('val11','val12','val13'),(val21,val22,val23),,(val31,val32,val33) and so forth like an array. This format saves me from implementing javascript iteration and i guess faster than multiple single inserts.